securesystemslib/signer/_signer.py

```diff
@@ -225,5 +252,294 @@ def sign(self, payload: bytes) -> Signature:
                 securesystemslib.exceptions.UnsupportedAlgorithmError:
                 Signing errors.
         """
-        sig_dict = sslib_keys.create_signature(self.key_dict, payload)
-        return Signature(**sig_dict)
+        return self._crypto_signer.sign(payload)
+
+
+class CryptoSigner(Signer, metaclass=ABCMeta):
```

Comment:
> Is SSlibSigner just a legacy component now?

Correct. I thought I mentioned this somewhere, but maybe I didn't. It looks like I only complained about the inconsistent and proprietary on-disk and in-memory data structure, which is the meat of SSlibSigner. I just couldn't find a way to fix it, without breaking the SSlibSigner interface.

So, yes, I do consider CryptoSigner to be a modern replacement for SSlibSigner use cases. And for a transition period, CryptoSigner can also provide the implementation for SSlibSigner, which would allow us to deprecate `key*` independently of SSlibSigner. I'm okay to revert the latter part, if that only makes things more complicated.

Regarding class hierarchy of CryptoSigner, I agree that the subclasses could be non-public API. The only reason why I made them public, is to namespace the corresponding generate methods: `RSASigner.generate`, etc. But we could also expose these directly on CryptoSigner, e.g. `CryptoSigner.generate_rsa`, etc.

Reply:
> > Is SSlibSigner just a legacy component now?
> 
> Correct. I thought I mentioned this somewhere, but maybe I didn't. 

Mentioned it in: https://github.com/secure-systems-lab/securesystemslib/pull/590#issuecomment-1638176113. Should have mentioned it in this PR too.